src/NuGet.Clients/NuGet.VisualStudio.Implementation/SolutionExplorer/Models/AssetsFileDependenciesSnapshot.cs

```diff
@@ -0,0 +1,214 @@
+using System.IO;
+using System.Linq;
+
+using Microsoft;
+
+using NuGet.Packaging;
+using NuGet.ProjectModel;
+
+namespace NuGet.VisualStudio.SolutionExplorer.Models
+{
+    /// <summary>
+    /// Snapshot of data captured from <c>project.assets.json</c>. Immutable.
+    /// </summary>
+    internal sealed class AssetsFileDependenciesSnapshot
+    {
+        /// <summary>
+        /// Gets the singleton empty instance.
+        /// </summary>
+        public static AssetsFileDependenciesSnapshot Empty { get; } = new AssetsFileDependenciesSnapshot(null, null);
+
+        /// <summary>
+        /// Shared object for parsing the lock file. May be used in parallel.
+        /// </summary>
+        private static readonly LockFileFormat LockFileFormat = new LockFileFormat();
+
+        public ImmutableDictionary<string, AssetsFileTarget> DataByTarget { get; }
+
+        /// <summary>
+        /// The <c>packageFolders</c> array from the assets file. The first is the 'user package folder',
+        /// and any others are 'fallback package folders'.
+        /// </summary>
+        private readonly ImmutableArray<string> _packageFolders;
+
+        /// <summary>
+        /// Lazily populated instance of a NuGet type that performs package path resolution. May be used in parallel.
+        /// </summary>
+        private FallbackPackagePathResolver? _packagePathResolver;
+
+        /// <summary>
+        /// Produces an updated snapshot by reading the <c>project.assets.json</c> file at <paramref name="path"/>.
+        /// If the file could not be read, or no changes are detected, the current snapshot (this) is returned.
+        /// </summary>
+        public AssetsFileDependenciesSnapshot UpdateFromAssetsFile(string path)
+        {
+            Requires.NotNull(path, nameof(path));
+
+            try
+            {
+                // Parse the file
+                using var fileStream = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read | FileShare.Delete, 4096 * 10, FileOptions.SequentialScan);
+
+                LockFile lockFile = LockFileFormat.Read(fileStream, path);
+
+                return new AssetsFileDependenciesSnapshot(lockFile, this);
+            }
+            catch
+            {
+                return this;
+            }
+        }
+
+        private AssetsFileDependenciesSnapshot(LockFile? lockFile, AssetsFileDependenciesSnapshot? previous)
+        {
+            if (lockFile == null)
+            {
+                DataByTarget = ImmutableDictionary<string, AssetsFileTarget>.Empty;
+                return;
+            }
+
+            Assumes.NotNull(previous);
+
+            _packageFolders = lockFile.PackageFolders.Select(pf => pf.Path).ToImmutableArray();
+
+            ImmutableDictionary<string, AssetsFileTarget>.Builder dataByTarget = ImmutableDictionary.CreateBuilder<string, AssetsFileTarget>(StringComparer.OrdinalIgnoreCase); // TODO review comparer here -- should it be ignore case?
```

Comment:
Who owns this TODO? We only allow TODO's with linked issues, so please create one against us and either provide details, or reference the blocking issue on another repo.

Reply:
Removed.